ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -41,12 +41,19 @@ def __init__(self, parent, node, indent=2):
         self.indent = indent
 
         self.join_tables = []
+        self.join_table_aliases = []
         self.join_types = []
         self.join_predicates = []
 
     def _translate(self, expr):
         return self.parent._translate(expr)
 
+    def record_join_tables(self, op: ops.Relation):
```

Comment:
@chloeh13q asked:

> would it be better to have key-value pairs as opposed to two separate lists?

The only place these two lists are used is here:

```python
        for jtype, table, alias, preds in zip(
                self.join_types, self.join_tables[1:],
                self.join_table_aliases[1:], self.join_predicates
        ):
```

We could certainly "encode" the implicit-index-based relation between these zipped lists in a more explicit data structure. But that would call for a slight refactoring of `TableSetFormatter`.

Perhaps, @cpcloud has opinions about this.